### PR TITLE
Support replicating delete_range request but do not delete data when applying

### PR DESCRIPTION
**Pull request by MyonKeminta**
## What have you changed? (mandatory)

This PR adds a field named `notify_only` to kv delete_range request and Raft delete_range request. `notify_only` is used to signal that a unsafe_destroy_range request is coming to delete data in the range. The data will not be deleted when applying the delete_range request with `notify_only` set.

## What are the type of the changes? (mandatory)

- New feature (change which adds functionality)

## How has this PR been tested? (mandatory)

- By unit tests

## Does this PR affect documentation (docs) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Refer to a related PR or issue link (optional)

* Changes to kvproto that should be merged first: pingcap/kvproto#373
* Changes to TiDB that need to be merged after this PR: pingcap/tidb#10069

### CODE REVIEW

**Comment by MyonKeminta** [comment]
image
Here is a simple test. A cluster with 3 nodes was created and I run sysbench oltp 128thread on it. During the time in the red box, I starts a client continuously sending delete_range(notify_only = true) to the cluster (with concurrency 3). It shows that there's no any obvious performance impact.

**Comment by MyonKeminta** [comment]
@overvenus @zhangjinpeng1987 PTAL

**Comment by overvenus** [comment]
What's the current status of this PR? Is it still in "DNM"?

**Comment by MyonKeminta** [comment]
@overvenus You can review it now. It's DNM because the corresponding PR in kvproto haven't been merged.

File: src/raftstore/store/fsm/apply.rs [hunk 1]
```diff
@@ -1270,38 +1271,43 @@ impl ApplyDelegate {
...
-                "{} failed to delete all in range [{}, {}), cf: {}, err: {:?}",
-                self.tag,
-                escape(&start_key),
-                escape(&end_key),
-                cf,
-                e
-            );
-        });
-
+        // TODO: Should this be executed when `notify_only` is set?
```
Comment (AndreMouche) [inline src/raftstore/store/fsm/apply.rs]: When would we remove the data?

**Comment by AndreMouche** [comment]
Please update the kvproto

File: src/raftstore/store/fsm/apply.rs (hunk 1 above)
Comment (MyonKeminta) [reply src/raftstore/store/fsm/apply.rs]: kvproto = { git = "https://github.com/pingcap/kvproto.git", branch="release-3.0" }

Comment (MyonKeminta) [reply src/raftstore/store/fsm/apply.rs]: Did you mean removing data in the range? They should be removed by invoking unsafe_destroy_range by TiDB.

**Comment by MyonKeminta** [comment]
@AndreMouche PTAL again thanks!

**Pull request merged**